components/script/dom/bindings/buffer_source.rs

```diff
@@ -52,15 +53,14 @@ use crate::script_runtime::{CanGc, JSContext};
 /// provides a view onto an `ArrayBuffer`.
 ///
 /// See: <https://webidl.spec.whatwg.org/#BufferSource>
-#[derive(PartialEq)]
 pub(crate) enum BufferSource {
     /// Represents an `ArrayBufferView` (e.g., `Uint8Array`, `DataView`).
     /// See: <https://webidl.spec.whatwg.org/#ArrayBufferView>
-    ArrayBufferView(Box<Heap<*mut JSObject>>),
+    ArrayBufferView(RootedTraceableBox<Heap<*mut JSObject>>),
```

Comment:
@jdm After the changes in https://github.com/servo/mozjs/pull/594, many Heap usages now need to be rooted. I fixed some cases by adding RootedTraceableBox, and added `#[cfg_attr(crown, allow(crown::unrooted_must_root))]` where the fix is more complex (e.g., in gpubuffer.rs). We can create a followup ticket to address those. Please review.